src/main/java/org/xbill/DNS/SVCBBase.java

```diff
@@ -185,20 +204,20 @@ public String toString() {
   }
 
   public static class ParameterAlpn extends ParameterBase {
-    private List<byte[]> values;
+    private final List<byte[]> values;
 
     public ParameterAlpn() {
       super();
       values = new ArrayList<>();
     }
 
-    public ParameterAlpn(List<byte[]> values) {
+    public ParameterAlpn(List<String> values) {
       super();
-      this.values = values;
+      this.values = values.stream().map(String::getBytes).collect(Collectors.toList());
```

Comment:
getBytes uses the platform default encoding. This is often UTF-8, but it's definitely not guaranteed, so unless it's always ASCII, there will be problems.